django_celery_beat/admin.py

```diff
@@ -138,9 +138,25 @@ class PeriodicTaskAdmin(admin.ModelAdmin):
         }),
     )
     readonly_fields = (
-        'last_run_at',
+        'last_run_at', 'crontab_translation',
     )
 
+    def crontab_translation(self, obj):
+        return obj.crontab.human_readable
+
+    change_form_template = 'admin/djcelery/change_periodictask_form.html'
+
+    def changeform_view(self, request, object_id=None, form_url='',
+                        extra_context=None):
+        extra_context = extra_context or {}
+        crontabs = CrontabSchedule.objects.all()
```

Comment:
instead of calling all, can we reduce the number?

Reply:
Hmmm, I'm not sure how we'd decide which to include. My goal here is to provide natural language translations for all available crontabs.

The best argument for reducing the number is possible slow response times when loading the page. But, based on other sites I've used that load a lot of data into javascript objects onto the DOM, my hunch is that there would have to be hundreds (if not thousands) of CrontabSchedule objects before it would impact performance of this page. Additionally, since this is an admin-only view, any hit to responsiveness would never be seen by end users.

Do you have thoughts on how we might choose which ones to reduce it to?

Reply:
can we show 20/25 per page?

Reply:
It's not clear to me how paginating this would work. 

My goal with this change is to update the human readable description anytime the user changes the select list of defined crontabs. So, if I change from `5 * * * *` to `0 10 * * 1`, I want the description to change from "At 5 minutes past the hour" to "At 10:00 AM, only on Monday" to verify that the change is what I expect.

If the code only loads 20 per page, what happens if I choose a crontab that wasn't part of the 20 that were loaded?

This could be implemented as ajax/XHR so that the description is readable only on demand. Since django-celery-beat doesn't currently have any custom endpoints defined, this seems like a bigger change. Alternately, there could be a setting to disable this if a site has so many CrontabSchedules defined that loading this admin page is onerous.

Can you tell me more about your objection to having all defined on this page?